homeassistant/components/conversation/default_agent.py

```diff
@@ -592,6 +610,80 @@ def _get_error_text(
         response_str = lang_intents.error_responses.get(response_key)
         return response_str or _DEFAULT_ERROR_TEXT
 
+    def register_trigger(
+        self,
+        sentences: list[str],
+        callback: core.CALLBACK_TYPE,
+        response: str | None = None,
+    ) -> core.CALLBACK_TYPE:
+        """Register a list of sentences that will trigger a callback when recognized."""
+        action = TriggerAction(callback, response=response)
+        with self._trigger_lock:
+            for sentence in sentences:
+                self._trigger_sentences[sentence].append(action)
+            self._rebuild_trigger_intents()
```

Comment:
When Home Assistant starts up, it will register a bunch of sentences and causing this rebuild to be called a lot. Can we instead rebuild this on demand when it's needed the first time ?